### PR TITLE
clean up boundary conditions

### PR DESCRIPTION
**Pull request by szy21**
# Description

Address #1417 

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by szy21** [comment]
> I might suggest reverting the `_sfc` suffix. I think these states represent the surface for some `bctype`s, but they represent other boundaries for different `bctype`s.

I don't have much preference. @akshaysridhar any suggestion? What are the `bctypes` where the states don't represent the surface?

File: src/Atmos/Model/bc_momentum.jl [hunk 1]
```diff
@@ -148,28 +148,28 @@ function atmos_momentum_normal_boundary_flux_second_order!(
...
+    state_sfc⁺,
+    diffusive_sfc⁺,
+    hyperdiffusive_sfc⁺,
+    aux_sfc⁺,
     bctype,
     t,
-    state1⁻,
-    diff1⁻,
-    aux1⁻,
+    state_int⁻,
```
Comment (akshaysridhar) [inline src/Atmos/Model/bc_momentum.jl:161]: Suggest adding a doc-string / comments to clarify these arguments; e.g. `_int` refers to first interior node; other than that this seems reasonable.

**Comment by szy21** [comment]
Sounds good. I will add the comments. What do we do for state_sfc? just state? state_bdy? (I'm making up abbreviations) If you don't have strong preference I could also keep it as state_sfc?

**Comment by charleskawczynski** [comment]
I think `state` is fine (these are in the `boundary_state` method after all), but I don't have a strong preference. I think `state_sfc` is fine if it makes sense to users at TOA-- I just wanted to make sure it was clear that the boundary state is for all boundaries and not just at altitude = 0.

**Comment by szy21** [comment]
> I think `state` is fine (these are in the `boundary_state` method after all), but I don't have a strong preference. I think `state_sfc` is fine if it makes sense to users at TOA-- I just wanted to make sure it was clear that the boundary state is for all boundaries and not just at altitude = 0.

Sounds good, I will revert it to state then.

**Comment by szy21** [comment]
bors try
